src/ngraph/runtime/cpu/pass/cpu_fusion.cpp

```diff
@@ -520,6 +520,68 @@ void ngraph::runtime::cpu::pass::CPUFusion::construct_conv_bias_bprop()
     this->add_matcher(m, callback);
 }
 
+static bool switch_nodes(std::shared_ptr<ngraph::Node> node1,
+                         std::shared_ptr<ngraph::Node> node2,
+                         size_t source_input_index = 0,
+                         size_t target_input_index = 0)
+{
+    // check if node1 has only 1 argument, not sure how it will work with >1 args
+    if (node1->inputs().size() > 1)
+    {
+        NGRAPH_DEBUG << "Cannot switch. More than 1 inputs to this node\n";
+        return false;
+    }
+    if (node1->get_users().size() > 1)
+    {
+        NGRAPH_DEBUG << "Cannot switch. More than 1 user of this node\n";
+        return false;
+    }
+    if (node1->outputs().size() > 1)
+    {
+        NGRAPH_DEBUG << "Cannot switch. More than 1 output of this node\n";
+        return false;
+    }
+    if (node2->outputs().size() > 1)
+    {
+        NGRAPH_DEBUG << "Cannot switch. More than 1 output of this node\n";
+        return false;
+    }
+
+    auto target_inputs = node2->output(0).get_target_inputs();
+
+    // actual switch happening after this
+    auto arg = node1->get_argument(source_input_index);
+    node2->input(0).replace_source_output(arg);
+
+    node1->input(0).replace_source_output(node2->output(0));
+
+    // used implementation ref from replace_node
+    for (auto& input : target_inputs)
+    {
+        input.replace_source_output(node1->output(0));
+    }
```

Comment:
In replace_node, we have "replacement->add_node_control_dependents(target); target->clear_control_dependents();". Here we also need to take care of control dependency.